packages/react-client/src/ReactFlightClient.js

```diff
@@ -1586,12 +1586,36 @@ function resolveErrorDev(
       'resolveErrorDev should never be called in production mode. Use resolveErrorProd instead. This is a bug in React.',
     );
   }
-  // eslint-disable-next-line react-internal/prod-error-codes
-  const error = new Error(
-    message ||
-      'An error occurred in the Server Components render but no message was provided',
-  );
-  error.stack = stack;
+
+  let error;
+  if (!enableOwnerStacks) {
+    // Executing Error within a native stack isn't really limited to owner stacks
+    // but we gate it behind the same flag for now while iterating.
+    // eslint-disable-next-line react-internal/prod-error-codes
+    error = Error(
+      message ||
+        'An error occurred in the Server Components render but no message was provided',
+    );
+    error.stack = stack;
+  } else {
+    const callStack = buildFakeCallStack(
+      response,
+      stack,
+      // $FlowFixMe[incompatible-use]
+      Error.bind(
+        null,
+        message ||
+          'An error occurred in the Server Components render but no message was provided',
+      ),
+    );
+    const rootTask = response._debugRootTask;
+    if (rootTask != null) {
+      error = rootTask.run(callStack);
```

Comment:
This doesn't actually have any effect. The async task doesn't get captured by the error. That might be a bug in `console.createTask`. The async task that's applied is whatever task that `reportError` runs in - which is whatever client component this threw inside of (`<Shell>` in this case).

We could potentially restore the task of the server component that errored somehow before calling `onUncaughtError` such as if the source of the error was a `React.lazy`. We could also potentially annotate the error object with the server component's task. Currently we don't track which component originally threw the error.